src/pivot-table/components/cells/DimensionTitleCell.tsx

```diff
@@ -1,38 +1,129 @@
+  translator: stardust.Translator;
+  changeSortOrder: ChangeSortOrder;
+  changeActivelySortedHeader: ChangeActivelySortedHeader;
 }
 
+const baseFlex: React.CSSProperties = {
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+};
+
+const labelWrapperStyle: React.CSSProperties = {
+  ...baseFlex,
+  flexDirection: "row",
+  overflow: "hidden",
+  minWidth: SORT_ICON_WIDTH + HEADER_TEXT_MIN_WIDTH,
+};
+
 const labelTextStyle: React.CSSProperties = {
   ...textStyle,
   fontWeight: "600",
-  alignSelf: "flex-end",
+  alignSelf: "center",
   flexGrow: 1,
+  paddingLeft: "8px",
+  minWidth: HEADER_TEXT_MIN_WIDTH,
 };
 
 export const testId = "title-cell";
 
-const DimensionTitleCell = ({ cell, style, isLastColumn }: LabelCellProps): JSX.Element => {
+const DimensionTitleCell = ({
+  cell,
+  style,
+  isLastColumn,
+  translator,
+  changeSortOrder,
+  changeActivelySortedHeader,
+}: DimensionTitleCellProps): JSX.Element => {
   const styleService = useStyleContext();
   const { fontSize, fontFamily } = styleService.header;
+  const anchorRef = useRef<HTMLDivElement>(null);
+
+  const isDim = cell.id !== "PSEUDO-DIM";
+
+  const headerData = useMemo<Header>(
+    () => ({
+      id: cell.id,
+      isDim,
+      fieldId: cell.fieldId,
+      qLibraryId: cell.qLibraryId,
+      label: "right",
+      headTextAlign: "right" as Align,
+      sortDirection: cell.sortDirection,
+      colIdx: cell.colIdx,
+      qReverseSort: cell.qReverseSort,
+      isActivelySorted: cell.isActivelySorted,
+    }),
+    [cell, isDim],
+  );
+
+  const sortFromMenu = async (evt: React.MouseEvent, newSortDirection: SortDirection) => {
+    evt.stopPropagation();
+    await changeSortOrder(headerData, newSortDirection);
+  };
 
   return (
     <div
-      title={cell}
+      title={cell.title}
       style={{
         ...style,
         ...getBorderStyle(true, isLastColumn, styleService.grid.border, false),
         ...styleService.header.rowTitle,
-        display: "flex",
+        padding: 0,
+        position: "relative",
+        display: "grid",
+        gridTemplateColumns: "1fr 24px",
+        gridGap: "4px",
+        minWidth: SORT_ICON_WIDTH + HEADER_TEXT_MIN_WIDTH + MENU_ICON_WIDTH,
       }}
       data-testid={testId}
     >
-      <div style={{ ...labelTextStyle, fontSize, fontFamily }}>{cell}</div>
+      <div style={{ ...labelWrapperStyle }}>
+        {cell.isActivelySorted && (
+          <div style={{ ...baseFlex, marginLeft: "8px" }}>
+            {cell.sortDirection === "A" ? <Ascending height="12px" /> : <Descending height="12px" />}
+          </div>
+        )}
+        <div style={{ ...labelTextStyle, fontSize, fontFamily }}>{cell.title}</div>
+      </div>
+
+      {isDim && (
+        <>
+          <HeadCellMenu
+            headerData={headerData}
+            translator={translator}
+            tabIndex={-1}
+            anchorRef={anchorRef}
+            handleHeadCellMenuKeyDown={() => console.log("keyDown")}
```

Comment:
`console.log`?

Reply:
I need to make it optional in shared-lib, for now i'll pass it a `() => {}` function